core/pkg/store/store.go

```diff
@@ -50,49 +52,77 @@ func (f *State) hasPriority(stored string, new string) bool {
 	return true
 }
 
-func NewFlags() *State {
-	return &State{
-		Flags:             map[string]model.Flag{},
+func NewStore(logger *logger.Logger) (*Store, error) {
+
+	schema := &memdb.DBSchema{
+		Tables: map[string]*memdb.TableSchema{
+			"flags": {
+				Name: "flags",
+				Indexes: map[string]*memdb.IndexSchema{
+
+					"id": {
+						Name:    "id",
+						Unique:  true,
+						Indexer: &memdb.StringFieldIndex{Field: "Key", Lowercase: false},
+					},
+				},
+			},
+		},
+	}
+
+	// Create a new data base
+	db, err := memdb.NewMemDB(schema)
+	if err != nil {
+		return nil, fmt.Errorf("unable to initialize flag database: %w", err)
+	}
+
+	return &Store{
 		SourceDetails:     map[string]SourceDetails{},
 		MetadataPerSource: map[string]model.Metadata{},
-	}
+		db:                db,
+		logger:            logger,
+	}, nil
 }
 
-func (f *State) Set(key string, flag model.Flag) {
-	f.mx.Lock()
-	defer f.mx.Unlock()
-	f.Flags[key] = flag
+// Deprecated: use NewStore instead
+func NewFlags() *Store {
+	state, err := NewStore(logger.NewLogger(nil, false))
+	if err != nil {
+		panic(fmt.Sprintf("unable to create flag store: %v", err))
+	}
+	return state
 }
 
-func (f *State) Get(_ context.Context, key string) (model.Flag, model.Metadata, bool) {
-	f.mx.RLock()
-	defer f.mx.RUnlock()
-	metadata := f.getMetadata()
-	flag, ok := f.Flags[key]
-	if ok {
-		metadata = f.GetMetadataForSource(flag.Source)
-	}
+func (f *Store) Get(_ context.Context, key string) (model.Flag, model.Metadata, bool) {
+	f.logger.Debug(fmt.Sprintf("getting flag %s", key))
+	txn := f.db.Txn(false)
 
-	return flag, metadata, ok
+	raw, err := txn.First("flags", "id", key)
+	flag, ok := raw.(model.Flag)
+	if err != nil || !ok {
+		return model.Flag{}, f.getMetadata(), false
+	}
+	return flag, f.GetMetadataForSource(flag.Source), true
 }
 
-func (f *State) SelectorForFlag(_ context.Context, flag model.Flag) string {
+func (f *Store) SelectorForFlag(_ context.Context, flag model.Flag) string {
 	f.mx.RLock()
 	defer f.mx.RUnlock()
 
 	return f.SourceDetails[flag.Source].Selector
 }
 
-func (f *State) Delete(key string) {
-	f.mx.Lock()
-	defer f.mx.Unlock()
-	delete(f.Flags, key)
-}
-
-func (f *State) String() (string, error) {
+func (f *Store) String() (string, error) {
+	f.logger.Debug(fmt.Sprintf("dumping flags to string"))
```

Comment:
Can you address the static check error, please?

Reply:
Yep, done.